lib/azure/armrest/configuration.rb

```diff
@@ -64,14 +64,9 @@ def self.cache_token(configuration)
       # Maximum number of threads to use within methods that use Parallel for thread pooling.
       attr_accessor :max_threads
 
-      # The environment in which to acquire your token.
-      attr_reader :environment
-
-      # The authority URL used to acquire a valid token.
-      attr_accessor :resource_url
-
-      # The resource URL used to acquire a valid token.
-      attr_accessor :authority_url
+      # The environment object which determines various endpoint URL's. The
+      # default is Azure::Armrest::Environment::Public.
+      attr_accessor :environment
 
       # Yields a new Azure::Armrest::Configuration objects. Note that you must
       # specify a client_id, client_key, tenant_id. The subscription_id is optional
```

Comment:
@bronaghs Not sure what you mean. It's *technically* optional in the Configuration constructor, but almost all Service classes require it. It's not required for the SubscriptionService class, though.